railties/lib/rails/generators/rails/app/templates/config/initializers/new_framework_defaults_7_0.rb.tt

```diff
@@ -83,11 +83,13 @@
 # Rails.application.config.active_storage.variant_processor = :vips
 
 # If you're upgrading and haven't set `cookies_serializer` previously, your cookie serializer
-# was `:marshal`. Convert all cookies to JSON, using the `:hybrid` formatter.
+# was `:marshal`. The default for new apps is `:json`.
 #
-# If you're confident all your cookies are JSON formatted, you can switch to the `:json` formatter.
+# You can convert all cookies to JSON using the `:hybrid` formatter. It is fine to use
+#`:hybrid` long term; you should do that unless you're confident that *all* your cookies
+# have been converted to JSON.
```

Comment:
I think some of the confusion here is that setting serialize to `:hybrid` doesn't convert
cookies unless users visit the site. 
Maybe something like: 
```suggestion
# To migrate an existing application to the `:json` serializer set this option to `:hybrid`.
# Rails transparently deserializes existing (Marshal-serialized) cookies on read and
# re-writes them in the JSON format. 
```
This is mostly rewording of the cookie guide:
https://guides.rubyonrails.org/action_controller_overview.html#cookies